include/seqan3/core/detail/reflection.hpp

```diff
@@ -0,0 +1,168 @@
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+#pragma once
+
+/*!\file
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ * \brief Static reflection for arbitrary types.
+ */
+
+#include <utility>
+
+#include <seqan3/range/container/constexpr_string.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Extracts the size of the display name of the given template.
+ * \tparam type The template parameter to get the size of the display name for.
+ * This functions uses gcc's magic [__PRETTY_FUNCTION__](https://gcc.gnu.org/onlinedocs/gcc/Function-Names.html)
+ * variable to get the size of the display name.
+ * This code adapts the snippet from the following
+ * [discussion](https://stackoverflow.com/questions/35941045/can-i-obtain-c-type-names-in-a-constexpr-way/35943472#35943472).
+ * \ingroup core
+ */
+template <typename type>
+struct get_display_name_size
+{
+    //!\brief Helper function to extract the size of the display name.
+    static constexpr auto _get_size = [] ()
+    {
+        // __PRETTY_FUNCTION__ exposes the signature of the class including the name of the template instance as
+        // a static const char[]. The following code, extracts the part that displays the template name.
+        auto name_ptr = __PRETTY_FUNCTION__;
+
+        while (*name_ptr++ != '=')
+        {}
+
+        for (; *name_ptr == ' '; ++name_ptr)
+        {}
+
+        char const* end_name_ptr = name_ptr;
+        int count = 1;
+
+        for (;;++end_name_ptr)
+        {
+            switch (*end_name_ptr)
+            {
+                case '[':
+                    ++count;
+                    break;
+                case ']':
+                    --count;
+                    if (!count)
+                        return size_t(end_name_ptr - name_ptr);
+            }
+        }
+        return size_t(0);
+    };
+
+    //!\brief Value constant storing the size of the display name of `type`.
+    //!\ingroup detail
```

Comment:
member variables (static or not) don't go into modules, because the belong to the class they are in